newIDE/app/src/AssetStore/AssetStoreSearchFilter.js

```diff
@@ -0,0 +1,168 @@
+  objectTypes: Set<string>;
+
+  constructor(objectTypes: Set<string> = new Set()) {
+    this.objectTypes = objectTypes;
+  }
+
+  isSatisfiedBy(searchItem: AssetShortHeader): boolean {
+    return (
+      this.objectTypes.size === 0 || this.objectTypes.has(searchItem.objectType)
+    );
+  }
+}
+
+export class LicenseAssetStoreSearchFilter
+  implements SearchFilter<AssetShortHeader> {
+  attributionFreeOnly: boolean;
+
+  static noAttributionLicenses = [
+    'CC0 (public domain)',
+    'Apache 2.0',
+    'Trademark of GDevelop Ltd (fair use authorised in GDevelop games)',
+    'SIL Open Font License 1.1 (OFL)',
+  ];
+
+  constructor(attributionFreeOnly: boolean = false) {
+    this.attributionFreeOnly = attributionFreeOnly;
+  }
+
+  isSatisfiedBy(searchItem: AssetShortHeader): boolean {
+    return (
+      !this.attributionFreeOnly ||
+      LicenseAssetStoreSearchFilter.noAttributionLicenses.includes(
+        searchItem.license
+      )
+    );
+  }
+}
+
+export class AnimatedAssetStoreSearchFilter
+  implements SearchFilter<AssetShortHeader> {
+  mustBeAnimated: boolean;
+  mustHaveSeveralState: boolean;
+
+  constructor(
+    mustBeAnimated: boolean = false,
+    mustHaveSeveralState: boolean = false
+  ) {
+    this.mustBeAnimated = mustBeAnimated;
+    this.mustHaveSeveralState = mustHaveSeveralState;
+  }
+
+  isSatisfiedBy(searchItem: AssetShortHeader): boolean {
+    const hasAnimatedState = searchItem.maxFramesCount > 1;
+    const hasSeveralState = searchItem.animationsCount > 1;
+    return (
+      (!this.mustBeAnimated || hasAnimatedState) &&
+      (!this.mustHaveSeveralState || hasSeveralState)
+    );
+  }
+}
+
+export class DimensionAssetStoreSearchFilter
+  implements SearchFilter<AssetShortHeader> {
+  dimensionMin: number;
+  dimensionMax: number;
+
+  static boundMin = 8;
+  static boundMax = 2048;
+
+  constructor(
+    dimensionMin: number = DimensionAssetStoreSearchFilter.boundMin,
+    dimensionMax: number = DimensionAssetStoreSearchFilter.boundMax
+  ) {
+    this.dimensionMin = dimensionMin;
+    this.dimensionMax = dimensionMax;
+  }
+
+  isSatisfiedBy(searchItem: AssetShortHeader): boolean {
+    return (
+      ((this.dimensionMin === DimensionAssetStoreSearchFilter.boundMin ||
+        this.dimensionMin <= searchItem.width) &&
+        (this.dimensionMin === DimensionAssetStoreSearchFilter.boundMax ||
+          searchItem.width <= this.dimensionMax)) ||
+      ((this.dimensionMin === DimensionAssetStoreSearchFilter.boundMin ||
+        this.dimensionMin <= searchItem.height) &&
+        (this.dimensionMin === DimensionAssetStoreSearchFilter.boundMax ||
+          searchItem.height <= this.dimensionMax))
+    );
+  }
+}
+
+/**
+ * Modulo operation (the remainder after dividing one number by another)
+ * @param x Dividend value.
+ * @param y Divisor value.
+ * @returns Return the remainder for the values.
+ */
+const mod = function(x: number, y: number): number {
```

Comment:
Why can't you use the remainder operation `%` ?

Reply:
I also prefer the formula with the double `%` but it's not a lot more simple and this method was used in the engine so I kept the same implementation.

Reply:
`-14 % 8 = -6` and `-14 mod % 8 = 2`, but I guess Clement meant a formula like `(x % y + x) % y` to avoid a call to floor.
It's true that it can't be negative but I think a mod make it more explicit.
uh, I copied pasted a documentation that is a bit confusing.